### PR TITLE
docs: update template option descriptions

### PR DESCRIPTION
**Pull request by fbiville**
... so that it's also reflected on the README

### CODE REVIEW

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 1]
```diff
@@ -39,8 +39,8 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
   @TemplateParameter.GcsReadFile(
       order = 2,
       optional = true,
-      description = "Path to the Neo4j connection metadata",
-      helpText = "The path to the Neo4j connection metadata JSON file.")
+      description = "Path to the Neo4j connection JSON file, hosted on GCS.",
```
Comment (fbiville) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:42]: I can also link to the schema if that helps (here and for the next option)
https://neo4j.com/docs/common-content-dataflow/0.1/_attachments/schemas/connection.v1.0.json

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 2]
```diff
@@ -30,7 +30,7 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
       order = 1,
       description = "Path to the job specification file",
       helpText =
-          "The path to the job specification file, which contains the configuration for source and target metadata.")
+          "The path to the job specification file, which contains the JSON description of data sources, Neo4j targets and actions.")
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:33]: I'm not sure if this should be:

The path to the job specification file, which contains the JSON description of data sources, Neo4j targets, and actions.

Or:

The path to the job specification file, which contains the JSON description of data sources as well as the Neo4j targets and actions.

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java (hunk 1 above)
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:42]:
```suggestion
      description = "The path to the Neo4j connection JSON file, which is hosted on Google Cloud Storage.",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 3]
```diff
@@ -49,9 +49,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
   @TemplateParameter.Text(
       order = 3,
       optional = true,
-      description = "Secret ID for the Neo4j connection metadata",
+      description =
+          "Secret ID for the Neo4j connection information. This is encoded as JSON (same schema as neo4jConnectionUri).",
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:53]:
```suggestion
          "The secret ID for the Neo4j connection information. This value is encoded in JSON, using the same schema as the `neo4jConnectionUri`.",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 4]
```diff
@@ -49,9 +49,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
...
       helpText =
-          "The secret ID for the Neo4j connection metadata. This is an alternative to the GCS path option.")
+          "The secret ID for the Neo4j connection metadata. This is an alternative to neo4jConnectionUri.")
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:55]:
```suggestion
          "The secret ID for the Neo4j connection metadata. You can use this value as an alternative to the `neo4jConnectionUri`.")
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 5]
```diff
@@ -60,9 +61,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
   @TemplateParameter.Text(
       order = 4,
       optional = true,
-      description = "Options JSON",
-      helpText = "Options JSON. Use runtime tokens.",
-      example = "{token1:value1,token2:value2}")
+      description =
+          "JSON object also called \"runtime tokens\", defining named values interpolated into the specification URIs and queries.",
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:65]:
```suggestion
          "A JSON object that defines named values interpolated into the specification URIs and queries. Also known as runtime tokens.",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 6]
```diff
@@ -60,9 +61,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
...
+      helpText = "JSON object also called \"runtime tokens\"",
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:66]:
```suggestion
      helpText = "A JSON object that is also called runtime tokens",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 7]
```diff
@@ -60,9 +61,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
...
+      example = "{token1:value1,token2:value2}. Spec can refer to $token1 and $token2.")
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:67]: What does "Spec" mean here?

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 8]
```diff
@@ -71,8 +73,8 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
   @TemplateParameter.Text(
       order = 5,
       optional = true,
-      description = "Query SQL",
-      helpText = "Override SQL query.")
+      description = "SQL query override for all BigQuery sources",
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:76]:
```suggestion
      description = "The SQL query override for all BigQuery sources.",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 9]
```diff
@@ -81,8 +83,8 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
   @TemplateParameter.GcsReadFile(
       order = 6,
       optional = true,
-      description = "Path to Text File",
-      helpText = "Override text file pattern",
+      description = "Text File Path override for all Text sources",
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:86]:
```suggestion
      description = "The text file path override for all text sources.",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 10]
```diff
@@ -81,8 +83,8 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
...
+      helpText = "Text File Path override",
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:87]:
```suggestion
      helpText = "The text file path override",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java (hunk 2 above)
Comment (fbiville) [reply v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:33]: The second phrasing definitely reads better

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java (hunk 7 above)
Comment (fbiville) [reply v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:67]: sorry, it's the import specification, whose URI is specified with the `jobSpecUri` option

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 11]
```diff
@@ -49,9 +50,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
   @TemplateParameter.Text(
       order = 3,
       optional = true,
-      description = "Secret ID for the Neo4j connection metadata",
+      description =
+          "The secret ID for the Neo4j connection information. This value is encoded in JSON, using the same schema as the `neo4jConnectionUri`.",
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:54]:
```suggestion
          "The secret ID for the Neo4j connection information. This value is encoded in JSON, using the same schema as the `neo4jConnectionUri` parameter.",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 12]
```diff
@@ -49,9 +50,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
...
       helpText =
-          "The secret ID for the Neo4j connection metadata. This is an alternative to the GCS path option.")
+          "The secret ID for the Neo4j connection metadata. You can use this value as an alternative to the `neo4jConnectionUri`.")
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:56]:
```suggestion
          "The secret ID for the Neo4j connection metadata. You can use this value as an alternative to the `neo4jConnectionUri` parameter.")
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 13]
```diff
@@ -60,9 +62,10 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
   @TemplateParameter.Text(
       order = 4,
       optional = true,
-      description = "Options JSON",
-      helpText = "Options JSON. Use runtime tokens.",
-      example = "{token1:value1,token2:value2}")
+      description =
+          "A JSON object that defines named values interpolated into the specification URIs and queries. Also known as runtime tokens.",
+      helpText = "A JSON object that is also called runtime tokens",
+      example = "{token1:value1,token2:value2}. Spec can refer to $token1 and $token2.")
```
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:68]:
```suggestion
      example = "{token1:value1,token2:value2}. The import specification can refer to $token1 and $token2.")
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java (hunk 2 above)
Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:33]:
```suggestion
          "The path to the job specification file, which contains the JSON description of data sources as well as the Neo4j targets and actions.")
```

Comment (rszper) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:33]:
```suggestion
          "The path to the job specification file, which contains the JSON description of data sources as well as the Neo4j targets and actions.")
```

**Review by rszper — APPROVED** [review]
One suggested edit, otherwise LGTM.

**Review by liferoad — APPROVED** [review]

**Pull request merged**